iroh-p2p/src/keys.rs

```diff
@@ -52,8 +52,9 @@ impl From<Keypair> for libp2p::identity::Keypair {
         match kp {
             Keypair::Ed25519(kp) => {
                 let mut bytes = kp.to_bytes();
-                let kp = libp2p::identity::ed25519::Keypair::decode(&mut bytes)
+                let kp = libp2p::identity::ed25519::Keypair::try_from_bytes(&mut bytes)
```

Comment:
Why do we expect this to be valid here is it user generated content from the other side of the conection?

Reply:
We do not expect it to be valid, hence the rename from `decode` to `try_from_bytes` as its clear now it can fail.